IPLD-Data-Model-v1.md

```diff
@@ -1,17 +1,103 @@
-# [WIP] IPLD Data Model
+IPLD Data Model
+===============
 
-## Simple Types
+The IPLD Data Model is "layer 1" of the IPLD core models.
+("Layer 0" are the [Codecs](Codecs); "layer 2" is called [Schemas](schema).)
+
+The Data Model specifies what categories of data are representable in IPLD --
+for example, booleans, and integers, and textual strings.
+[Codecs](Codecs) specify exactly how these data are transcribed into serialized bytes.
+
+
+Kinds
+-----
+
+The following is the list of essential _kinds_ (or more formally, _representation kinds_)
+of data representable in the IPLD Data Model:
 
-* Boolean
 * Null
+* Boolean
+* Integer
+* Float
 * String
+* Bytes
+* List
+* Map
+* Link
+
+(Note that we use the term "_kinds_" here to disambiguate this from "_types_",
+which is a term we'll use at the [Schemas](schema) level.)
+
+The _recursive kinds_ are:
+
+* List
+* Map
+
+The _scalar kinds_ (the complement of recursive) are:
+
+* Null
+* Boolean
 * Integer
 * Float
-* Array
-* Object (Hash Map)
-* Binary
+* String
+* Bytes
+* Link
+
+(Note that [Schemas](schema) introduce a few more kinds -- when clarification is necessary,
+these Data Model kinds can be called the "_representation kinds_",
+while the additional kinds introduced in the Schema layer are "_perceived kinds_".)
+
+
+### Null kind
+
+Null is a scalar kind.  Its cardinality is one -- the only value is 'null'.
+
+### Boolean kind
 
-## Link Type
+Boolean is a scalar kind.  Its cardinality is two -- either the value 'true' or the value 'false'.
 
-This type represents a link to another IPLD Block. The link reference
+### Integer kind
+
+
+### Float kind
+
+
+### String kind
+
+String is a scalar kind.  Its cardinality is infinite -- strings do not have a length limit.
+
+Strings are generally understood to be UTF-8 text, in [NFC](https://www.unicode.org/reports/tr15/#Norm_Forms) canonicalization.
+
+### Bytes kind
+
+Bytes is a scalar kind.  Its cardinality is infinite -- byte sequences do not have a length limit.
+
+Bytes are distinct from strings in that they are not considered to have any character encoding nor
+generally expected to be printable as human-readable text.
+In order to print byte sequences as text, addition effort such as Base64 encoding may be required.
+
+### List kind
+
+List is a recursive kind.
+
+Values contained in lists can be accessed by their ordinal offset in the list.
+
```

Comment:
This area could probably be improved by lifting @rvagg's "Singular" vs "Associative" terminology from https://github.com/ipld/specs/pull/110/files...

Reply:
There's something a bit off here and in Map about using the description of _what you can do_ with them rather than _what they are_. Since a List and Map can hold arbitrary other kinds, do we have to build up the concept from first principles rather than just saying "a List is like an array" and "a Map is like an associative array"?